src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -430,6 +437,40 @@ void DynamoViewModelRequestAboutWindow(DynamoViewModel model)
             aboutWindow.ShowDialog();
         }
 
+        private void OnGalleryBackgroundMouseClick(object sender, MouseButtonEventArgs e)
+        {
+            dynamoViewModel.CloseGalleryCommand.Execute(null);
+            e.Handled = true;
+        }
+
+        void DynamoViewModelRequestShowGallery()
+        {
+            if (galleryView == null) //On-demand instantiation
+            {
+                galleryView = new GalleryView(new GalleryViewModel(dynamoViewModel));
+                Grid.SetColumnSpan(galleryBackground, mainGrid.ColumnDefinitions.Count);
+                Grid.SetRowSpan(galleryBackground, mainGrid.RowDefinitions.Count);
+            }
+
+            if (GalleryViewModel.HasContents)
```

Comment:
Now we need the instance of `GalleryViewModel` inside `GalleryView` to access this `HasContents` property, so we can expose a new property called `GalleryView.ViewModel` to access the `GalleryViewModel`.